### PR TITLE
Functional tests - Add test 'Enable delivery time in stock products'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test 'Enable delivery time in stock products'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts" URL_FO=shopUrl/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18436)

### CODE REVIEW

**Review by SimonGrn — DISMISSED** [review]

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.productSettingsLink,
+    );
+    const pageTitle = await this.pageObjects.productSettingsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+  });
+
+  const tests = [
+    {args: {action: 'enable', enable: true, deliveryTimeText: '3-4 days'}},
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js:57]:
```suggestion
    {args: {enable: true, deliveryTimeText: '3-4 days'}},
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    {args: {action: 'disable', enable: false, deliveryTimeText: ''}},
+  ];
+  tests.forEach((test, index) => {
+    it(`should ${test.args.action} delivery time of in-stock products`, async function () {
+      await testContext.addContextItem(this, 'testIdentifier', `${test.args.action}StockManagement`, baseContext);
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js:62]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', `setDeliveryTimeInStocks_${index}`, baseContext);
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const pageTitle = await this.pageObjects.productSettingsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.productSettingsPage.pageTitle);
+  });
+
+  const tests = [
+    {args: {action: 'enable', enable: true, deliveryTimeText: '3-4 days'}},
+    {args: {action: 'disable', enable: false, deliveryTimeText: ''}},
+  ];
+  tests.forEach((test, index) => {
+    it(`should ${test.args.action} delivery time of in-stock products`, async function () {
```
Comment (nesrineabdmouleh) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js:61]:
```suggestion
    it('should set delivery time of in-stock products', async function () {
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/03_productsStock/06_enableDeliveryTimeInStockProducts.js:57]: why do you want to delete it,  this is used to know which action is done before the check. and without it, we can't know if the checks are after enabling or disabling

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js [hunk 4]
```diff
@@ -197,4 +198,15 @@ module.exports = class productSettings extends BOBasePage {
     await this.clickAndWaitForNavigation(this.savePaginationFormButton);
     return this.getTextContent(this.alertSuccessBlock);
   }
+
+  /**
+   * Set delivery time text
+   * @param deliveryTimeText
+   * @return {Promise<string>}
+   */
+  async setDeliveryTimeInStocks(deliveryTimeText) {
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]:
```suggestion
  async setDeliveryTimeInStock(deliveryTimeText) {
```

File: tests/puppeteer/pages/FO/product.js [hunk 5]
```diff
@@ -132,4 +136,20 @@ module.exports = class Product extends FOBasePage {
...
+   */
+  isDeliveryInformationVisible() {
+    return this.elementVisible(this.deliveryInformationSpan, 1000);
+  }
+
+  /**
+   * Get delivery information text
+   * @return {Promise<string>}
+   */
+  isDeliveryInformationText() {
```
Comment (Progi1984) [inline tests/puppeteer/pages/FO/product.js]:
```suggestion
  getDeliveryInformationText() {
```

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]: done thanks

File: tests/puppeteer/pages/FO/product.js (hunk 5 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/FO/product.js]: done thanks

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**